vizier/backend/src/info/vizierdb/commands/plot/BarChart.scala

```diff
@@ -35,64 +35,87 @@ import info.vizierdb.artifacts.VegaAutosize
 import info.vizierdb.artifacts.VegaPadding
 import info.vizierdb.artifacts.VegaLegend
 import info.vizierdb.artifacts.VegaLegendType
+import info.vizierdb.artifacts.VegaValueReference
+import info.vizierdb.artifacts.VegaTitle
 
-object BarChart extends Command
-{
-  val PARAM_SERIES = "series" 
+
+object BarChart extends Command {
+  val PARAM_SERIES = "series"
   val PARAM_DATASET = "dataset"
   val PARAM_X = "xcol"
   val PARAM_Y = "ycol"
   val PARAM_FILTER = "filter"
   val PARAM_COLOR = "color"
   val PARAM_LABEL = "label"
   val PARAM_ARTIFACT = "artifact"
+  val PARAM_Y_AXIS = "yList"
 
   override def name: String = "Bar Chart"
 
   override def parameters: Seq[Parameter] = Seq(
-    ListParameter(id = PARAM_SERIES, name = "Bars", components = Seq(
-      DatasetParameter(id = PARAM_DATASET, name = "Dataset"),
-      ColIdParameter(id = PARAM_X, name = "X-axis"),
-      ColIdParameter(id = PARAM_Y, name = "Y-axis"),
-      StringParameter(id = PARAM_LABEL, name = "Label", required = false),
-      StringParameter(id = PARAM_FILTER, name = "Filter", required = false, helpText = Some("e.g., state = 'NY'")),
-      StringParameter(id = PARAM_COLOR, name = "Color", required = false, helpText = Some("e.g., #214478")),
-    )),
-    StringParameter(id = PARAM_ARTIFACT, name = "Output Artifact (blank to show only)", required = false)
+    ListParameter(
+      id = PARAM_SERIES,
+      name = "Bars",
+      components = Seq(
+        DatasetParameter(id = PARAM_DATASET, name = "Dataset"),
+        ColIdParameter(id = PARAM_X, name = "X-axis"),
+        ListParameter(id = PARAM_Y_AXIS, name = "Y-axes", components = Seq(ColIdParameter(id = PARAM_Y, name = "Y-axis"))),
+        NumericalFilterParameter(id = PARAM_FILTER, name = "Filter",required = false),
+        StringParameter(id = PARAM_LABEL, name = "Label", required = false),
```

Comment:
Note previously proposed changes to NumericalFilter and Color.

Reply:
Also, something that's not at all clear here: the Label and Color parameters are both part of a list... there's one per dataset, per y-axis label, per category value.  Here, we only have one per dataset.  It seems like it's not possible to use these parameters, as-is.

It seems like you want another ListParameter of Key/Label/Color triples.